### PR TITLE
FCN-352 cursor hook: auto-format agent edits with prettier

### PR DESCRIPTION
**Pull request by vishsanghishetty**
## Description

Adds a Cursor IDE hook that runs Prettier on files right after the AI agent edits them. Right now, formatting only gets enforced at commit time via lint-staged, which means the agent's raw output can look messy in the editor until you commit. This hook closes that gap — files get formatted the moment the agent writes them.

Two files:
- `.cursor/hooks.json` — wires up the `afterFileEdit` event
- `.cursor/hooks/format-on-edit.sh` — reads the filepath from stdin, checks the extension, runs `npx prettier --write` on supported types (ts, tsx, js, jsx, mjs, json, css, scss, md)

The script gracefully exits 0 on everything — unknown file types, missing files, empty input. It never blocks the agent.

This works alongside our existing husky + lint-staged setup. lint-staged is the safety net at commit; this hook is the real-time cleanup while you're working.

**Jira issue #** FCN-352

**Backport of** #

## Type of Change

- [x] Configuration change
- [x] New feature/enhancement (non-breaking change which adds functionality)

### Manual Testing

**Test Steps:**
1. Created a badly-formatted `.tsx` file and ran the hook script manually via `echo '{"path":"src/_test-format-hook.tsx"}' | .cursor/hooks/format-on-edit.sh`
2. Verified the file was properly formatted after hook execution (diff confirmed formatting applied)
3. Tested edge cases: empty input, non-matching file type (.png), missing file — all exit 0 gracefully
4. Verified `jq` and `prettier` dependencies are available in the repo

**Test Environment:**
- [x] Tested locally

### Automated Testing

N/A — this is a shell script hook, not a component. Tested manually with multiple edge cases.

## Screenshots/Recordings

N/A — no UI changes.

### Code Quality
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have removed any console logs and debugging code
- [x] My changes generate no new warnings or errors

### Documentation
- [x] I have updated the documentation accordingly

### Dependencies
- [x] Any dependent changes have been merged and published

## Breaking Changes

**Does this PR introduce breaking changes?**
- [x] No

## Additional Notes

The hook only runs inside Cursor IDE — it has zero impact on CI, other editors, or non-Cursor users. The `.cursor/hooks/` directory is already in the repo (we have `.cursor/rules/` and `.cursor/skills/` committed too), so this follows the existing pattern.

This is the team's first committed Cursor hook. It sets the pattern for future agent-aware tooling we might add.

### Focus Areas
- Does the file extension list cover what we need? Currently: ts, tsx, js, jsx, mjs, json, css, scss, md
- Is the error handling sufficient (always exit 0, stderr suppressed)?

### Questions for Reviewers
- Should we add any other file types to the format list?

### CODE REVIEW

File: .cursor/hooks/format-on-edit.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+# runs prettier on files the agent just edited.
+# lint-staged catches this at commit time too, but this keeps
+# the diff clean in real-time while the agent works.
+
+input=$(cat)
+filepath=$(echo "$input" | jq -r '.path // empty')
```
[the remark this answers is not in the document]
Comment (vishsanghishetty) [reply .cursor/hooks/format-on-edit.sh]: Good catch — fixed in 366a220. The payload uses `file_path` not `path`, so the hook was silently doing nothing. Updated the jq extraction to `.file_path // empty`.

**Review by davidaznaur — APPROVED** [review]

**Comment by kelvah** [comment]
Are we sure we want to push a script for that?
You can configure the same thing with a couple of clicks inside cursor vs code settings.

**Comment by vishsanghishetty** [comment]
> Are we sure we want to push a script for that? You can configure the same thing with a couple of clicks inside cursor vs code settings.

yeah that's fair Roberto, cursor does have `editor.formatOnSave` which covers most of this. the difference is that the hook only triggers on agent edits (not manual saves), and since it's committed to the repo every dev who clones gets it automatically without having to configure anything locally. but honestly a shared .vscode/settings.json with `formatOnSave: true` is simpler and lower maintenance, and covers both agent and manual edits. i'll close this one and we can go that route instead if the team wants it.
